app/Models/Foundation/Summit/ProposedSchedule/SummitProposedSchedule.php

```diff
@@ -197,6 +197,32 @@ public function getScheduledSummitEventsByLocationAndDateRange(
         return $this->scheduled_summit_events->matching($criteria)->toArray();
     }
 
+    /**
+     * @param \DateTime $date
+     * @param SummitAbstractLocation|null $location
+     * @return SummitProposedScheduleSummitEvent[]
+     */
+    public function getProposedPublishedEventsBeforeThan(\DateTime $date, SummitAbstractLocation $location):array {
+
+        $criteria = Criteria::create();
+        $criteria->where(Criteria::expr()->lt('end_date', $date));
+        $criteria->andWhere(Criteria::expr()->eq('location', $location));
+        return $this->scheduled_summit_events->matching($criteria)->toArray();
+    }
+
+    /**
+     * @param \DateTime $date
+     * @param SummitAbstractLocation|null $location
+     * @return SummitProposedScheduleSummitEvent[]
+     */
+    public function getProposedPublishedEventsAfterThan(\DateTime $date, SummitAbstractLocation $location):array {
```

Comment:
this is not the asked method
please read the SDS
at publishing time ( on proposed schedule) , we need to get the previous and the next published activity at the same location

https://docs.google.com/document/d/1cHQk0ITDrNICMtjluMBdapaEEMw2odUJeH0FEECgRvU/edit

Reply:
done